### DISCUSSION TITLE
Add a  Bean lifecycle/PostConstruct for beans created by micronaut

### DISCUSSION THREAD

**Discussion by subash89** [micronaut-projects/micronaut-core#9431] [Q&A] [answered]
Hi, 
I am trying to create some additional beans off of some bean created by micronaut (Jdbi). For every datasource, micronaut creates a jdbi. How can I get a listener or a hook for a bean that's not create by me. (_The end goal here is to create an additional bean for each jdbi bean created)._
 I am using micronaut 3.9.x latest.

**Comment by yawkat** [comment, marked as answer]
check out BeanCreatedEventListener
